### ISSUE TITLE
Wrong tables deactivated when a table is removed

### ISSUE THREAD

**Issue by thufir-hawat** [metabase/metabase#6953]
Hey, I want to use Metabase with a mysql db but apparently it can't display any tables.

https://imgur.com/7By2geJ

My setup is as follows:

+-------------+
| @@Version |
+-------------+
| 5.5.58           |
+-------------+

+-----------------------+
| Database                    |
+-----------------------+
| information_schema   |
| mysql                          |
| performance_schema |
+-----------------------+

Chrome: Version 64.0.3282.167
DB runs on Ubuntu: Version 12.04.03
Metabase latest runs on Win10 (.jar started with Powershell)

So, there's only one db available here, yet Metabase still can't display any tables.

**Comment by jornh** [comment]
What’s the Metabase.jar log telling you?

Specially when you do a scan/sync of the database.

**Comment by thufir-hawat** [comment]
After Sync & Scan:

02-16 12:19:00 [1mDEBUG metabase.middleware[0m :: POST /api/database/2/sync_schema 200 (896 Ás) (1 DB calls)
02-16 12:19:00 [1mDEBUG sync.util[0m :: Sync operations in flight: {:sync-metadata #{2}}
02-16 12:19:00 [1mINFO sync.util[0m :: STARTING: Sync metadata for mysql Database 2 'DB'
02-16 12:19:01 [1mINFO sync.util[0m :: FINISHED: Sync metadata for mysql Database 2 'DB' (60 ms)
02-16 12:19:05 [1mDEBUG metabase.middleware[0m :: POST /api/database/2/rescan_values 200 (2 ms) (1 DB calls)
02-16 12:19:05 [1mDEBUG sync.util[0m :: Sync operations in flight: {:cache-field-values #{2}}
02-16 12:19:05 [1mINFO sync.util[0m :: STARTING: Cache field values in mysql Database 2 'DB'
02-16 12:19:05 [1mINFO sync.util[0m :: FINISHED: Cache field values in mysql Database 2 'DB' (9 ms)

**Comment by jornh** [comment]
OK, so that's consistent with that you're not seeing any tables. No errors - but also no 

What tables were you expecting to see (I assume you have them in the database you show above called `mysql`)- can you show the output of `mysql> use mysql; ` followed by `mysql> show tables;`?

**Comment by thufir-hawat** [comment]
Here's the output:

mysql> show tables;
+-----------------------------+
| Tables_in_mysql             |
+-----------------------------+
| _PRODDB                     |
| auth_group                  |
| auth_group_permissions      |
| auth_permission             |
| auth_user                   |
| auth_user_groups            |
| auth_user_user_permissions  |
| columns_priv                |
| cr_cr                       |
| cr_cr_backup                |
| cr_jirastatus               |
| cr_jirasubtasktemplate      |
| cr_jiratasktemplate         |
| cr_progressstatus           |
| cr_remainingeffort          |
| cr_remainingeffortcw        |
| cr_remainingeffortcw_backup |
| cr_systemerrorlog           |
| cr_test                     |
| cr_testprotocol             |
| cr_testspecification        |
| ctf_categories              |
| ctf_componentgroups         |
| ctf_components              |
| ctf_contacts                |
| ctf_contacts_categories     |
| ctf_defaultproperties       |
| ctf_dependencies            |
| ctf_hubs                    |
| ctf_release                 |
| ctf_servertype              |
| db                          |
| defect_defect               |
| django_admin_log            |
| django_content_type         |
| django_migrations           |
| django_session              |
| estimate_estimate           |
| event                       |
| func                        |
| general_log                 |
| help_category               |
| help_keyword                |
| help_relation               |
| help_topic                  |
| host                        |
| hour_whour                  |
| hour_whour_backup           |
| import_cr_neu               |
| logger_generallog           |
| ndb_binlog_index            |
| plugin                      |
| proc                        |
| procs_priv                  |
| protocol_protocol           |
| proxies_priv                |
| release_component           |
| release_component_backup    |
| release_deployment          |
| release_release             |
| release_release_backup      |
| release_release_components  |
| sdpd_cache_table            |
| servers                     |
| servreq_servreq             |
| slow_log                    |
| tables_priv                 |
| time_zone                   |
| time_zone_leap_second       |
| time_zone_name              |
| time_zone_transition        |
| time_zone_transition_type   |
| topic_topic                 |
| topic_topicbooking          |
| user                        |
+-----------------------------+
76 rows in set (0.01 sec)

**Comment by jornh** [comment]
OK, this is weird ... here are a few things to try (but you're more than welcome to add your own ideas):

* And (of course 🙂!) you're using the same username/password when using the `mysql` client and Metabase?
* **Follow the really nice https://metabase.com/docs/latest/troubleshooting-guide/datawarehouse.html** 

In addition:
* Can you post a screenshot of the Metabase Admin Database screen for your DB? (Tip: just drag and drop your screenshot to GitHub or use the ☁️ upload button, no need for imgur! 👍)
* Try setting up a new connection besides your existing `DB` - collect and check the Metabase log for this action
* Can you run Metabase directly on the Ubuntu machine - and get it working?

**Comment by thufir-hawat** [comment]
Thanks for the tips.

This seems to be the same issue as mine: https://github.com/metabase/metabase/issues/5887

I can reach the DB with a native query, so the connection definitely works. 

Here's the admin db screen (is that the right one?):
![db-admin](https://user-images.githubusercontent.com/35777154/36315486-2a9b0d7a-1338-11e8-9fa4-19cad4f43ce4.PNG)

I only have access to MySQL here and I already dumped the db and set it up anew - no success. The built in db works, though.

I don't have access to the server, just the db... and I tried it with another similar MySQL db as well, same result (or lack thereof).

**Comment by jornh** [comment]
OK, good points ... even tough we've not gotten to to bottom of it - good find with #5887 

Regarding the admin screen, you're close: I meant if you click into `DB` - I'm specially interested in if you put `mysql` into the 'database name:' field there.

I'm not really sure if #5887 talks about a config where `INFORMATION_SCHEMA` is put into that field. Because I wouldn't expect that to work. For comparison I have (an old v0.26.0.RC2 Metabase) configured with Database name: `poc` matching this in the `mysql` native client:

```
mysql> select @@version;
+-----------+
| @@version |
+-----------+
| 5.7.17    |
+-----------+
1 row in set (0.00 sec)

mysql> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| poc                |
+--------------------+
2 rows in set (0.00 sec)

mysql> use poc;
Database changed
mysql> show tables;
+---------------------+
| Tables_in_poc       |
+---------------------+
| Account             |
| AccountHashedClone  |
...
```

.... and that one is working as expected!

**Comment by thufir-hawat** [comment]
Oh, yeah, I did that:

![asd](https://user-images.githubusercontent.com/35777154/36319379-0db98ca8-1343-11e8-8398-eec7563ba11f.png)

I also tried changing the user, etc. The port is also standard (3306). 

Well, I don't really know what to do anymore. I'd like to switch to Postgresql but that's not an option, sadly. 😞

**Comment by dannyeuu** [comment]
I think this may be duplicated of #6825

**Comment by camsaul** [comment]
Fixed by #9638
